tests/test_core.py

```diff
@@ -171,6 +170,136 @@ def test_mass(Q, T):
     npt.assert_almost_equal(left, right)
 
 
+@pytest.mark.parametrize("Q, T", test_data)
+def test_mass_nan(Q, T):
+    T[1] = np.nan
+    m = Q.shape[0]
+
+    left = np.linalg.norm(
+        core.z_norm(core.rolling_window(T, m), 1) - core.z_norm(Q), axis=1
+    )
+    left[np.isnan(left)] = np.inf
+
+    right = core.mass(Q, T)
+    npt.assert_almost_equal(left, right)
+
+
+@pytest.mark.parametrize("Q, T", test_data)
+def test_mass_inf(Q, T):
+    T[1] = np.inf
+    m = Q.shape[0]
+
+    left = np.linalg.norm(
+        core.z_norm(core.rolling_window(T, m), 1) - core.z_norm(Q), axis=1
+    )
+    left[np.isnan(left)] = np.inf
+
+    right = core.mass(Q, T)
+    npt.assert_almost_equal(left, right)
+
+
+def test_apply_exclusion_zone():
```

Comment:
I think it is probably important to test the "edge cases: where `index = 0` and `index = 9`